src/analyzer/plugins/analyzerkeyfinder.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+#include <keyfinder/keyfinder.h>
+
+#include <QObject>
+
+#include "analyzer/plugins/analyzerplugin.h"
+#include "util/types.h"
+
+namespace mixxx {
+
+class AnalyzerKeyFinder : public AnalyzerKeyPlugin {
+  public:
+    static AnalyzerPluginInfo pluginInfo() {
```

Comment:
I still prefer to hide string constants as implementation details in the .cpp file.

- The method does not need to be defined inline here. Static methods could also be defined in the .cpp file.
- The string literals could be moved into an anonymous namespace as constants in the .cpp file to create them only once.

Reply:
I don't understand how either of those would be helpful. For 1, AnalyzerKey would need to #include analyzerkeyfinder.cpp... which seems wrong. For 2, why does it matter, the string literals are only used once and need to be accessible to this static function.

Reply:
No, you don't need to include the .cpp file. The constant are stored in the data segment section and efficiently returned as implicitly shared (= copy-on-write) QString.

Header files should only contain the bare minimum definitions required for compilation. The content of those strings is syntactically irrelevant and the strings are not declared as constexpr. That's why the should not be exposed in the header file.

Reply:
This is just a function that returns some data.

Reply:
I agree with @uklotzde. If we ever need to change this string, it would require rebuilding of all object files that include this header filer. If it definition is in the cpp file, only that single file needs to be recompiled.

Reply:
Done 37a7f58